tests/e2e/otel_tracing_test.go

```diff
@@ -183,3 +183,152 @@ func TestOTELTracingWithConsoleExporter(t *testing.T) {
 
 	t.Log("OTEL environment variables and header attribute args successfully verified in extProc container")
 }
+
+// TestOTELTracingWithGatewayConfig verifies that OTEL environment variables
+// can be configured via GatewayConfig and are properly injected into extProc containers.
+func TestOTELTracingWithGatewayConfig(t *testing.T) {
+	ctx, cancel := context.WithTimeout(t.Context(), 5*time.Minute)
+	defer cancel()
+
+	// Get the source directory relative to this test file.
+	_, filename, _, ok := runtime.Caller(0)
+	require.True(t, ok, "failed to get source file path")
+	sourceDir := filepath.Dir(filename)
+
+	manifest := filepath.Join(sourceDir, "testdata", "otel_tracing_gateway_config.yaml")
```

Comment:
Instead of doing this, delete this custom file and point directly to the file in the `examples` directory. This is what I meant in my previous comment. You already spotted an issue where the example was missing the `kubernetes` keyword, and we want to keep examples always working. Let's do that by using those in the e2e tests, as we do with other examples.